code/datums/components/defibrillator.dm

```diff
@@ -238,37 +243,39 @@
 
-	var/defib_success = TRUE
-
 	// Run through some quick failure states after shocking.
 	var/time_dead = world.time - target.timeofdeath
 
+	var/failure_message
 	if(!target.is_revivable())
-		user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Heart tissue damage beyond point of no return for defibrillation.</span>")
-		defib_success = FALSE
+		failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Heart tissue damage beyond point of no return for defibrillation.</span>"
 	else if(target.getBruteLoss() >= 180 || target.getFireLoss() >= 180)
-		user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Severe tissue damage detected.</span>")
-		defib_success = FALSE
+		failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Severe tissue damage detected.</span>"
 	else if(HAS_TRAIT(target, TRAIT_HUSK))
-		user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Subject is husked.</span>")
-		defib_success = FALSE
+		failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Subject is husked.</span>"
 	else if(target.blood_volume < BLOOD_VOLUME_SURVIVE)
-		user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Patient blood volume critically low.</span>")
-		defib_success = FALSE
+		failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Patient blood volume critically low.</span>"
 	else if(!target.get_organ_slot("brain"))  // So things like headless clings don't get outed
-		user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - No brain detected within patient.</span>")
-		defib_success = FALSE
+		failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - No brain detected within patient.</span>"
 	else if(ghost)
 		if(!ghost.can_reenter_corpse || target.suiciding) // DNR or AntagHUD
-			user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - No electrical brain activity detected.</span>")
+			failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - No electrical brain activity detected.</span>"
+		else
+			failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Patient's brain is unresponsive. Further attempts may succeed.</span>"
+	else if(HAS_TRAIT(target, TRAIT_FAKEDEATH))
+		if(signal_result & COMPONENT_DEFIB_FAKEDEATH_DENIED)
+			failure_message = "<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Patient's brain is unresponsive. Further attempts may succeed.</span>"
+		else if(signal_result & COMPONENT_DEFIB_FAKEDEATH_ACCEPTED)
+			// as much as I hate that this is here, it has to come after the `Patient is not in a valid state. Operation aborted.` check.
```

Comment:
This seems fine to be here? Am I missing something?

Reply:
Ideally there would only be a True and false condition, but theres 3 conditions here